client/src/containers/App/App.js

```diff
@@ -0,0 +1,171 @@
+import React, { Component } from 'react';
+import './App.css';
+import PropTypes from 'prop-types'
+import {
+    Collapse,
+    Navbar,
+    NavbarToggler,
+    NavbarBrand,
+    Nav,
+    NavItem,
+    Container,
+    Row,
+    Col,
+    Jumbotron,
+    Button
+  } from 'reactstrap';
+
+import {
+    Route,
+    Link
+  } from 'react-router-dom'
+
+import { userLogin, fetchEvents, postEvent } from '../../actions';
+
+import { connect } from 'react-redux';
+
+const Home = () => (
+<div>
+    <h2>Home</h2>
+</div>
+)
+
+const About = () => (
+<div>
+    <h2>About</h2>
+</div>
+)
+
+
+class App extends Component {
+
+    static propTypes = {
+        userState: PropTypes.object.isRequired,
+        events: PropTypes.object.isRequired,
+        appState: PropTypes.object.isRequired,
+        dispatch: PropTypes.func.isRequired
+    }
+
+    constructor() {
+        super();
+        this.state = {
+            isOpen: false
+        };
+
+        this.toggle = this.toggle.bind(this);
+    }
+
+    componentDidMount() {
+        this.getData();
+    }
+
+    toggle() {
+        this.setState({
+            isOpen: !this.state.isOpen
+        });
+    }
+
+    getData() {
+        const { dispatch } = this.props;
+        dispatch(userLogin({ username: 'admin',password: '1299459ML' }));
+        dispatch(fetchEvents());
```

Comment:
what do you think about separating actions dispatch? When I was working in react and redux we create mapDispatchToProps function and pass that function as a 2nd argument to connect function. I think it good to separate dispatch logic.

Reply:
I saw that pattern, didnt see whats the purpose - i only saw we needed to add an additional function because of that. Can you outline which dispatch logic we are separating?

Reply:
creating mapDispatchToProps function will map actions to props so we can easily send then to child or sub components.

Reply:
Why cant we send them to child components using this way?